src/model/delta/insertdelta.js

```diff
@@ -95,7 +104,44 @@ register( 'insert', function( position, nodes ) {
 	delta.addOperation( insert );
 	this.document.applyOperation( insert );
 
+	// When element is a DocumentFragment we need to move its markers to Document#markers.
+	if ( nodes instanceof DocumentFragment ) {
+		for ( const marker of nodes.markers ) {
+			const doc = this.document;
+			const range = moveRange( marker[ 1 ], position );
+			const markerDelta = new MarkerDelta();
+			const markerOperation = new MarkerOperation( marker[ 0 ], null, range, doc.markers, doc.version );
+
+			this.addDelta( markerDelta );
+			markerDelta.addOperation( markerOperation );
+			doc.applyOperation( markerOperation );
```

Comment:
> You can call `this.setMarker` instead of doing it manually. It should be safer.

This is a good point I wanted to point the same (before I got the doubts I described above :P)